src/ResumeLog.C

```diff
@@ -202,7 +202,7 @@ void parse_resumelog(std::vector<RunGroup *> &groups)
       assert(runstate_int >= 0 && runstate_int < NUM_RUNSTATES);
       runstate = (test_runstate_t) runstate_int;
 
-      res = fscanf(f, "%d\n", &result);
+      res = fscanf(f, "%d\n", reinterpret_cast<int*>(&result));
```

Comment:
result is an enum of unknown size.  It would be safer to make a new int variable to use in `fscanf` without the cast, and then cast the the int to the enum.  The `res != 1` would be clearer using the enumeration: `res != PASSED`.

Reply:
>result is an enum of unknown size. It would be safer to make a new int variable to use in fscanf without the cast, and then cast the the int to the enum.

Fair enough. Updated.

>The res != 1 would be clearer using the enumeration: res != PASSED.

`res` is checking if `fscanf` failed. We could leave that and add the punned `result` from the fscanf.

```c++
  int tmp{};
  res = fscanf(f, "%d\n", &tmp);
  if (res != 1) {
     result = CRASHED;
  } else {
    result = static_cast<test_results_e>(tmp);
  }
```